src/ResourceManager/Common/Commands.ScenarioTests.ResourceManager.Common/EnvironmentSetupHelper.cs

```diff
@@ -176,6 +181,110 @@ public void SetupEnvironment(AzureModule mode)
 #endif
         }
 
+        private void SetEnvironmentVariableFromCredentialFile()
+        {
+            var filePath = Environment.GetFolderPath(Environment.SpecialFolder.UserProfile) + "/.azure/testcredentials.json";
+            Dictionary<string, object> credentials;
+            using (StreamReader r = new StreamReader(filePath))
+            {
+                string json = r.ReadToEnd();
+                credentials = JsonUtilities.DeserializeJson(json);
+            }
+
+            if (Environment.GetEnvironmentVariable("TEST_CSM_ORGID_AUTHENTICATION") == null)
+            {
+                var formattedConnectionString = String.Format("SubscriptionId={0};HttpRecorderMode={1};Environment={2}", credentials["SubscriptionId"], credentials["HttpRecorderMode"], credentials["Environment"]);
```

Comment:
Use a StringBuilder / StriungBuilder.AppendFormat here, it will save space.  

Alternately, you could build a list of strings and use string.Join to join them using semicolons